mozilla/kuma
# bug 1287529 remove soft delete on document

Remove the soft deletion pattern (tombstone) used on MDN Documents in favor of hard deletion.

The feature in itself according to my testing works, however I don't have an S3 bucket attached to my local MDN which prevents me from testing the proper deletions of attachments.

**Testing steps:**
I used the following steps to test this pull request.

1. Create a document
2. Delete the document
3. Check for a deletion log
4. Check using `shell_plus` that the document is effectively non-present anymore in the database
5. Re-create the document
6. Re-delete the document

**Note:**
- There is still some tests in error, that at this stage I do not manage to fix, this PR is still a **work in progress**
- The commits in this PR are not descriptive, the PR should be rebased and the commit edited before merging.
